docs/source/config_file.rst

```diff
@@ -65,234 +65,346 @@ unfortunate, and is subject to change in future versions.
    The ``warn_unused_configs`` flag may be useful to debug misspelled
    section names.
 
-Global flags
-************
+.. note::
 
-The following global flags may only be set in the global section
-(``[mypy]``).
+   Configuration flags are liable to change between releases.
 
-- ``python_version`` (string) specifies the Python version used to
-  parse and check the target program.  The format is ``DIGIT.DIGIT``
-  for example ``2.7``.  The default is the version of the Python
-  interpreter used to run mypy.
 
-- ``platform`` (string) specifies the OS platform for the target
-  program, for example ``darwin`` or ``win32`` (meaning OS X or
-  Windows, respectively).  The default is the current platform as
-  revealed by Python's ``sys.platform`` variable.
+Examples
+********
 
-- ``always_true`` (comma-separated list of strings) gives variable
-  names that will be treated as compile-time constants that are always
-  true.
+Here is an example of a ``mypy.ini`` file.
 
-- ``always_false`` (comma-separated list of strings) gives variable
-  names that will be treated as compile-time constants that are always
-  false.
+.. code-block:: ini
 
-- ``custom_typing_module`` (string) specifies the name of an
-  alternative module which is to be considered equivalent to the
-  ``typing`` module.
+    [mypy]
+    python_version = 2.7
+    warn_return_any = True
+    warn_unused_configs = True
 
-- ``custom_typeshed_dir`` (string) specifies the name of an
-  alternative directory which is used to look for stubs instead of the
-  default ``typeshed`` directory.
+    [mypy-mycode.foo.*]
+    disallow_untyped_defs = True
 
-- ``mypy_path`` (string) specifies the paths to use, after trying the paths
-  from ``MYPYPATH`` environment variable.  Useful if you'd like to keep stubs
-  in your repo, along with the config file.
+    [mypy-mycode.bar]
+    warn_return_any = False
 
-- ``warn_incomplete_stub`` (Boolean, default False) warns for missing
-  type annotation in typeshed.  This is only relevant in combination
-  with ``disallow_untyped_defs`` or ``disallow_incomplete_defs``.
+    [mypy-somelibrary]
+    ignore_missing_imports = True
 
-- ``warn_redundant_casts`` (Boolean, default False) warns about
-  casting an expression to its inferred type.
+If you place this file at the root of your repo and run mypy, mypy will:
```

Comment:
I wonder if each bullet should just also quote the text from the example (e.g. `python_version = 2.7`) which it describes, to make it clear what the descriptions refer to?

Reply:
I tried doing this, but unfortunately it made the explanations look a little cluttered. I decided instead to group the explanations about the global options into one group and the per-module options into another. This way, only needs to read three bullet points at a time, instead of 6 -- hopefully that helps.